src/Apps/Collect2/Components/SeoProductsForCollection.tsx

```diff
@@ -0,0 +1,132 @@
+export function simpleConversionToUSD(price: number, currencyCode: string) {
+  if (currencyCode === "USD") {
+    return price
+  } else if (currencyCode === "GBP") {
+    return price * 1.28
+  } else if (currencyCode === "EUR") return price * 1.1
+}
+
+export const getMaxMinPrice = priceableArtworks => {
+  const priceRange = [null, null]
+  priceableArtworks.map(artwork => {
+    if (artwork.node.is_price_range && artwork.node.listPrice) {
+      const { maxPrice, minPrice } = artwork.node.listPrice
+      const convertedPrice = {
+        maxPrice: simpleConversionToUSD(maxPrice.major, maxPrice.currencyCode),
+        minPrice: simpleConversionToUSD(minPrice.major, minPrice.currencyCode),
+      }
+      if (!priceRange[1] || priceRange[1] < convertedPrice.maxPrice)
+        priceRange[1] = convertedPrice.maxPrice
+      if (!priceRange[0] || priceRange[0] > convertedPrice.minPrice)
+        priceRange[0] = convertedPrice.minPrice
+    } else if (artwork.node.listPrice && artwork.node.listPrice.major) {
+      const convertedPrice = simpleConversionToUSD(
+        artwork.node.listPrice.major,
+        artwork.node.listPrice.currencyCode
+      )
+      if (!priceRange[1] || convertedPrice > priceRange[1])
+        priceRange[1] = convertedPrice
+      if (!priceRange[0] || convertedPrice < priceRange[0])
+        priceRange[0] = convertedPrice
+    }
+  })
+  return priceRange
+}
+
+export const getFilteredArtworks = artworks_connection => {
+  const filtedWorks =
+    artworks_connection &&
+    artworks_connection.edges.filter(edge => {
+      return (
+        edge.node.listPrice &&
+        edge.node.sale_message !== "Contact For Price" &&
+        edge.node.sale_message !== "Sold"
+      )
+    })
+  return filtedWorks
+}
+
+export class SeoProducts extends React.Component<
+  SeoProductsForCollectionProps
+> {
+  render() {
+    const {
+      artworks: { artworks_connection },
+      collectionDescription,
+      collectionName,
+      collectionURL,
+    } = this.props
+
+    const priceAbleArtworks = getFilteredArtworks(artworks_connection)
+
+    return (
+      <>
+        <Product
+          data={{
+            name: collectionName,
+            description: collectionDescription,
+            url: collectionURL,
+            offers: {
+              "@type": "AggregateOffer",
+              lowPrice: getMaxMinPrice(priceAbleArtworks)[0],
+              highPrice: getMaxMinPrice(priceAbleArtworks)[1],
+              priceCurrency: "USD",
+              availability: "https://schema.org/InStock",
+            },
+          }}
+        />
+      </>
+    )
+  }
+}
+
+export const SeoProductsForCollection = createFragmentContainer(SeoProducts, {
+  artworks: graphql`
+    fragment SeoProductsForCollection_artworks on FilterArtworks
+      @argumentDefinitions(
+        first: { type: "Int", defaultValue: 30 }
+        after: { type: "String", defaultValue: "" }
+      ) {
+      artworks_connection(first: $first, after: $after) {
```

Comment:
Artist metadata uses [these parameters](https://github.com/artsy/reaction/blob/713a65c548d8c76421d17c80c13822c1687d44fc/src/Apps/Artist/Components/ArtistMeta.tsx#L258-L260) to make sure works are published and for sale. Maybe it could be used here as well? 

With that I think `getFilteredArtworks` would not be needed.

Reply:
@sepans Thanks for the feedback. 🤝 
I made some changes based on your feedback. 
You mentioned I should use  [these parameters](https://github.com/artsy/reaction/blob/713a65c548d8c76421d17c80c13822c1687d44fc/src/Apps/Artist/Components/ArtistMeta.tsx#L258-L260), but that one is from `artist`, this case is using `filterArtworks`. From the `filterArtworks`, it only has `sort, after, first, before, last`.
I am not familiar with Relay. Should I make a new query? or I did try to modify the query, but I got some errors. Something like, `said it must have the same query as his parent has`. It seems I can not modify the query.

Reply:
Yes you are absolutely right. sorry about misleading you 😓 😆

Reply:
@sepans 
It's ok.🙇 Thanks a lot.
I have another idea. I try to use two fragments. One is sorted by price (desc), and another is sorted by price(asc). Then I just get the first index of each one. When I try to modify Relay, I got many errors. It might need more time to figure it out. 
The existing PR, and the idea I mentioned above. Comparing two ideas. Is it worth to figure it out? or This PR is good enough. There are two reasons that I came up with another idea. 
First,  I saw there is a ticket on our board, it about improving the performance on the collection page. I guess this PR might hurt the performance.
Second, I found out the query only gets 30 artworks, not all of them. If there are more than 30 artworks on the page, the result is wrong.
What do you think?